integration-test/common/conftest.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+import os
+import tempfile
+import shutil
+import subprocess
+
+RUN_SCRIPT_FNAME = "run.sh"
+OUTPUT_DIR = "output_dir"
+SWEEP_GENERATOR = "generate_design_sweeps.py"
+
+def is_master(config):
+  """Return true we're on a master node."""
+  return not hasattr(config, 'workerinput')
+
+def pytest_addoption(parser):
+  parser.addoption("--sweep-file", action="store", help="Sweep file")
+
+def pytest_configure(config):
+  """Generate sweeps using the sweep generator.
+
+  Only the master node performs the sweep generation.
+  """
+  if is_master(config):
+    cwd = os.getcwd()
+    config.tmp_output_dir = tempfile.mkdtemp(dir=cwd)
+    _, config.tmp_sweep_file = tempfile.mkstemp(suffix=".xe", dir=cwd)
+    # Read the sweep file and update the `output_dir` field with a temp dir.
+    sweep_file = config.getoption("--sweep-file")
+    if not os.path.isabs(sweep_file):
+      sweep_file = os.path.join(cwd, sweep_file)
+    with open(sweep_file, "r") as f_in, open(config.tmp_sweep_file,
+                                             "w") as f_out:
+      fmt_key = "%%(%s)s" % OUTPUT_DIR
+      for line in f_in:
+        if fmt_key in line:
+          f_out.write(line % {OUTPUT_DIR: config.tmp_output_dir})
+        else:
+          f_out.write(line)
+    # Generate the sweeps using the generator.
+    sweeps_dir = "%s/../../sweeps" % os.environ["ALADDIN_HOME"]
+    os.chdir(sweeps_dir)
+    sweep_generator = os.path.join(sweeps_dir, SWEEP_GENERATOR)
+    process = subprocess.Popen(
```

Comment:
I don't know how easy this would be to do, but would it be possible to expose an API to `generate_design_sweeps.py` so that you could just import it as a library and run it without having to go through a subprocess? It's all Python in the end.

Reply:
I'm afraid the relative paths will be a problem.

Reply:
Oh yeah, it's not even in the same repository...well here's another reason why we should consider moving integration-tests to the gem5-aladdin repo instead.